code/game/events/naturify.dm

```diff
@@ -0,0 +1,334 @@
+		var/obj/structure/hanging_lantern/HL = new /obj/structure/hanging_lantern(L.loc)
+		HL.dir = L.dir
+		HL.lantern_can_be_removed = FALSE
+		HL.update()
+		qdel(L)
+
+	for(var/obj/machinery/door/airlock/AL in target)
+		if(!istype(AL, /obj/machinery/door/airlock/external))
+			new /obj/machinery/door/mineral/wood/log(AL.loc)
+			qdel(AL)
+	for(var/obj/machinery/door/unpowered/shuttle/S in target)
+		new /obj/machinery/door/mineral/wood/log(S.loc)
+		qdel(S)
+	for(var/obj/machinery/door/poddoor/P in target)
+		qdel(P)
+
+	if(spawn_flora)
+		for(var/turf/simulated/floor/F in target)
+			if(!F.has_dense_content() && prob(35))
+				var/flora_type = pickweight(flora_types)
+				new flora_type(F)
+
+	if(spawn_trees)
+		for(var/turf/simulated/floor/F in target)
+			if(!F.has_dense_content() && prob(5))
+				for(var/obj/O in F)
+					qdel(O)
+				new/obj/structure/flora/tree/shitty(F)
+
+	if(spawn_animals)
+		for(var/turf/simulated/floor/F in target)
+			if(!F.has_dense_content() && prob(4))
+				var/animal_type = pickweight(animal_types)
+				new animal_type(F)
+
+	if(spawn_medicine_cows)
+		for(var/turf/simulated/floor/F in target)
+			if(!F.has_dense_content() && prob(8))
+				var/mob/living/simple_animal/cow/medical_cow = generate_medicine_cow()
+				medical_cow.forceMove(F)
+
+
+/// Turns a room into a cave with rocks. Perfect for a caveman.
+/proc/caveify_room(var/area/target)
+	for(var/turf/T in target)
+		if(istype(T, /turf/simulated/floor/))
+			T.ChangeTurf(/turf/simulated/floor/asteroid/air)
+		else if(istype(T, /turf/simulated/wall) || istype(T, /turf/simulated/wall/r_wall))
+			T.ChangeTurf(/turf/unsimulated/mineral/random/air, tell_universe = 1)
+
+	for(var/turf/simulated/floor/F in target)
+		if(!F.has_dense_content() && prob(25))
+			var/cave_decor_type = pickweight(cave_decor_types)
+			new cave_decor_type(F)
+
+	for(var/obj/machinery/light/L in target)
+		var/obj/structure/hanging_lantern/HL = new /obj/structure/hanging_lantern/dim(L.loc)
+		HL.dir = L.dir
+		HL.lantern_can_be_removed = FALSE
+		HL.update()
+		qdel(L)
+
+/proc/generate_bear_den(var/area/target)
+	for(var/turf/simulated/floor/F in target)
+		if(!F.has_dense_content() && prob(15))
+			new /mob/living/simple_animal/hostile/bear(F)
+
+/// Does various things to make the room look old and run down. For instance, breaks machines, eliminates power, etc.
+/proc/break_room(var/area/target)
+	for(var/obj/machinery/power/apc in target)
+		qdel(apc)
+
+	for(var/obj/machinery/M in target)
+		if(!istype(M, /obj/machinery/computer) && prob(50))
+			M.stat |= BROKEN
+
+	for(var/obj/machinery/camera/C in target)
+		C.deactivate(null)
+
+	// Recursive check to uncharge all cells. Bit laggy!
+	for(var/turf/T in target)
+		uncharge_all_cells_recursive(T)
+
+/proc/uncharge_all_cells_recursive(var/atom/A)
+	var/obj/item/weapon/cell/C = A
+	if(istype(C))
+		C.charge = 0
+	for(var/atom/content in A.contents)
+		uncharge_all_cells_recursive(content)
+	A.update_icon()
```

Comment:
I think there's an admin verb in the secrets panel to do this already if you just want to call that; though it might only apply to SMES.

Reply:
Yeah, something as comprehensive as this does not exist.